### DISCUSSION TITLE
How the intricate network interactions and configurations when multiple Docker containers, hosted on different hosts, communicate with each other through a custom overlay network using swarm mode.

### DISCUSSION THREAD

**Discussion by rlabinc** [community/community#62984] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Please provide a detailed explanation of how the overlay network handles routing, ingress, and egress traffic while ensuring security and load balancing for the containers spread across a geographically distributed cluster in Docker swarm mode.

**Comment by origamiofficial** [comment]
Docker swarm mode uses an overlay network, creating an encrypted VXLAN tunnel between nodes, utilizing an ingress load balancer for external traffic. The Docker Engine handles routing based on IPVS, ensuring secure communication among containers while optimizing load balancing across the cluster.

**Comment by rlabinc** [comment]
Thanks!

**Comment by queenofcorgis** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
